nemo/collections/vlm/neva/model/base.py

```diff
@@ -570,18 +575,41 @@ def forward(
         elif isinstance(num_media_tiles, list):
             num_media_tiles = torch.tensor(num_media_tiles, dtype=torch.int, device=input_ids.device)
 
-        # Preprocess input, labels and loss mask.
-        combined_embeddings, final_labels, final_loss_mask, final_attention_mask = self._preprocess_data(
-            media_embeddings,
-            language_embeddings,
-            input_ids,
-            loss_mask,
-            labels,
-            use_inference_kv_cache,
-            media_token_index,
-            num_media_tiles,
-            attention_mask,
-        )  # [combined_seq_len, b, h_language], [b, combined_seq_len], [b, combined_seq_len]
+        if self.config.is_llava_next:
+            media_embeddings = torch.split(media_embeddings, num_media_tiles.tolist(), dim=0)
+            media_embeddings, feature_lens = pack_image_features(
+                media_embeddings,
+                image_sizes,
+                vision_feature_select_strategy='default',
+                image_newline=self.image_newline,
+            )
+            combined_embeddings, attention_mask, position_ids, final_labels, final_input_ids, final_loss_mask = (
+                merge_input_ids_with_image_features(
```

Comment:
I don't like this design here, reason - 
this method is very similar function as `self._preprocess_data`. In self._preprocess_data we do more things to support sp and cp. It will be good if we can re-use the same method. Can you tell me how these 2 methods are different?